bitnami/shared/kyverno/kafka-client-policies.yaml

```diff
@@ -0,0 +1,117 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: generate-kafka-certificate
+  annotations:
+    policies.kyverno.io/title: Create certificate for Kafka client
+    policies.kyverno.io/category: Sample
+    policies.kyverno.io/subject: Deployment, Statefulset
+    policies.kyverno.io/description: >-
+      This policy creates the kafka client certificate for the annotated targets.
+spec:
+  # Generate for existing targets
+  generateExisting: true
+  background: true
+  rules:
+    - name: generate-msk-certificate
+      match:
+        any:
+          - resources:
+              annotations:
+                uw.systems/msk-kafka-client: "*"
+              kinds:
+                - Deployment
+                - StatefulSet
+      generate:
+        # Delete certificate if the target is deleted
+        synchronize: true
+        # Keep the generated certificates if the policy is deleted
+        orphanDownstreamOnPolicyDelete: true
+        apiVersion: cert-manager.io/v1
+        kind: Certificate
+        name: "{{request.object.metadata.name}}-gen-kafka-cert"
+        namespace: "{{request.namespace}}"
+        data:
+          spec:
+            commonName: "{{request.namespace}}/{{request.object.metadata.annotations.\"uw.systems/msk-kafka-client\"}}"
```

Comment:
So, `commonName` (which takes place of dnsNames for this certificate) uses the provided annotation `uw.systems/msk-kafka-client`.
Does that mean, value of `uw.systems/msk-kafka-client` should be the same as the pod name in order for certificate to work?